File: ssTools/kukijson/c++/kukijson.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <jsoncpp/json/json.h>

std::string formatData(const Json::Value& jsonData) {
    std::string result;

    const Json::Value& namaArray = jsonData["nama"];
    const Json::Value& isiArray = jsonData["isi"];

    // Periksa apakah namaArray dan isiArray memiliki jumlah elemen yang sama
    if (namaArray.size() != isiArray.size()) {
        return result;
    }

    for (Json::ArrayIndex i = 0; i < namaArray.size(); ++i) {
        const std::string& nama = namaArray[i].asString();
        const std::string& isi = isiArray[i].asString();

        result += nama + "=" + isi;

        // Tambahkan tanda ';' setiap kecuali elemen terakhir
        if (i != namaArray.size() - 1) {
            result += ";";
        }
    }

    return result;
}

int main(int argc, char* argv[]) {
    std::string filename;

    // Periksa apakah jumlah argumen valid
    if (argc != 3) {
        std::cout << "Penggunaan: program.exe -f <nama_file.json>\n";
        return 1;
    }

    // Periksa apakah argumen pertama adalah -f
    if (std::string(argv[1]) != "-f") {
        std::cout << "Penggunaan: program.exe -f <nama_file.json>\n";
        return 1;
    }

    // Ambil nama file dari argumen kedua
    filename = argv[2];

    // Buka file JSON
    std::ifstream file(filename);
    if (!file.is_open()) {
        std::cout << "Gagal membuka file: " << filename << '\n';
        return 1;
    }

    // Baca isi file JSON
    std::string jsonData((std::istreambuf_iterator<char>(file)),
    std::istreambuf_iterator<char>());

    // Parse data JSON
    Json::Value root;
    Json::Reader reader;
    if (!reader.parse(jsonData, root)) {
        std::cout << "Gagal memparse JSON\n";
        return 1;
    }

    // Ubah data JSON ke string dengan format yang diinginkan
    std::string formattedData = formatData(root);
    if (formattedData.empty()) {
        std::cout << "Format data tidak valid\n";
        return 1;
    }

    // Tampilkan hasil
    std::cout << formattedData << '\n';

    // Tutup file
    file.close();

    return 0;
}

File: ssTools/kukijson/rust/kukijson.rs
use std::env;
use std::fs;
use serde_json::{self, Value};

fn main() {
    // Ambil argumen dari baris perintah
    let args: Vec<String> = env::args().collect();

    // Periksa apakah argumen yang diberikan sesuai dengan format yang diharapkan
    if args.len() != 3 || args[1] != "-f" {
        println!("Penggunaan: cargo run -- -f <nama_file.json>");
        return;
    }

    let filename = &args[2];

    // Baca file JSON
    let json_data = match fs::read_to_string(filename) {
        Ok(data) => data,
        Err(_) => {
            println!("Gagal membaca file: {}", filename);
            return;
        }
    };

    // Memparsing JSON
    let data: Value = match serde_json::from_str(&json_data) {
        Ok(value) => value,
        Err(_) => {
            println!("Gagal memparse JSON");
            return;
        }
    };

    // Ubah data JSON ke string dengan format yang diinginkan
    let formatted_data = format_data(&data);
    println!("{}", formatted_data);
}

fn format_data(data: &Value) -> String {
    let nama_array = &data["nama"];
    let isi_array = &data["isi"];

    // Periksa apakah nama_array dan isi_array memiliki jumlah elemen yang sama
    if let (Some(nama), Some(isi)) = (nama_array.as_array(), isi_array.as_array()) {
        if nama.len() == isi.len() {
            let formatted_entries: Vec<String> = nama.iter().zip(isi.iter())
                .map(|(nama, isi)| format!("{}={}", nama, isi))
                .collect();
            let formatted_string = formatted_entries.join(";");
            return formatted_string.replace("\"", ""); // Menghilangkan tanda kutip ganda
        }
    }

    String::new()
}
